dev/src/cubegen.cpp

```diff
@@ -563,6 +569,12 @@ nfr("cg_load_vox", "name", "S", "R:voxels]S?",
             auto p = buf.c_str() + 8;
             bool chunks_skipped = false;
             Voxels *voxels = nullptr;
+            map<int32_t, int32_t> node_graph;
```

Comment:
map is an expensive data structure, are the indices so sparse that its necessary (over `vector`)?

Reply:
there is no official documentation that specifies any requirements on the range or sparseness of these ids. a quick look at the file does show small mostly sequential integers.
given the shortlivedness of these structures and the low usage it shouldn't be measurable of a difference ?

Reply:
would unordered_map help a lot ? The datastructure i'm used to use here is backed by two vectors, one is a hashtable, holding an offset into the second vector holding the value avoiding per value allocations. For trivial value types folded down ofc into a single vector

Reply:
`unordered_map` are per lookup if you have a lot of keys, but their fixed overhead may well be even greater than `map`..

We load a LOT of .vox files, so each of those constructing and destructing 5 maps seems excessive. Of course, hard to tell until you measure. If they're small ints I'd vote for at least using `vector`, and I wonder if it can be done even more efficiently than that.

Anyway, you can merge it with `map` if you want to just move on, and note it in a `FIXME` and maybe a future task to optimize if you're bored with other things :)